packages/core/src/user-store/postgres/store.ts

```diff
@@ -89,60 +90,45 @@ export class PostgresUserStore implements UserStore {
-              case "SCALAR": {
-                tableBuilder = tableBuilder.addColumn(
-                  field.name,
-                  gqlScalarToSqlType[field.scalarTypeName],
-                  (col) => {
-                    if (field.notNull) col = col.notNull();
-                    return col;
-                  }
-                );
-                break;
-              }
-              case "ENUM": {
-                tableBuilder = tableBuilder.addColumn(
-                  field.name,
-                  "text",
-                  (col) => {
-                    if (field.notNull) col = col.notNull();
-                    col = col.check(
-                      sql`${sql.ref(field.name)} in (${sql.join(
-                        field.enumValues.map((v) => sql.lit(v))
-                      )})`
-                    );
-                    return col;
-                  }
-                );
-                break;
-              }
-              case "LIST": {
-                tableBuilder = tableBuilder.addColumn(
-                  field.name,
-                  "text",
-                  (col) => {
-                    if (field.notNull) col = col.notNull();
-                    return col;
-                  }
-                );
-                break;
-              }
-              case "RELATIONSHIP": {
-                tableBuilder = tableBuilder.addColumn(
-                  field.name,
-                  gqlScalarToSqlType[field.relatedEntityIdType.name],
-                  (col) => {
-                    if (field.notNull) col = col.notNull();
-                    return col;
-                  }
-                );
-                break;
-              }
+        Object.entries(this.schema!.tables).map(async ([tableName, table]) => {
+          const dbTableName = `${tableName}_${this.versionId}`;
+          let tableBuilder = tx.schema.createTable(dbTableName);
+          Object.entries(table).forEach(([columnName, column]) => {
+            // Handle scalar list columns
+            if (column.list) {
+              tableBuilder = tableBuilder.addColumn(
+                columnName,
+                "text",
+                (col) => {
+                  if (!column.optional) col = col.notNull();
+                  return col;
+                }
+              );
+            } else if (isEnumType(column.type)) {
+              // Handle enum types
+              tableBuilder = tableBuilder.addColumn(
+                columnName,
+                "text",
+                (col) => {
+                  if (!column.optional) col = col.notNull();
+                  col = col.check(
+                    sql`${sql.ref(columnName)} in (${sql.join(
+                      schema!.enums[column.type.slice(5)].map((v) => sql.lit(v))
+                    )})`
+                  );
+                  return col;
+                }
+              );
+            } else {
+              // Handle all other column types
```

Comment:
It might be worth adding a foreign key constraint here for "references" fields. That would be the SQL thing to do. This way, if someone tries to insert a `Share` entity with a `traderId` value that is not found in the `Trader` table `id` column, they'll get an error.

We weren't doing this before, but it's worth thinking about.